front/src/common/redux/auth/auth.saga.js

```diff
@@ -19,7 +21,23 @@ function* changePasswordSaga(payload) {
         if (payload === 'is success') {
             yield put({ type: types.CHANGE_PASSWORD_STATE, });
         } else {
-            const response = yield call(changePasswordApi, payload);
+            let response;
+            const { oldPassword, newPassword } = payload.user;
+
+            if (payload.user.hasOwnProperty('teacherId') && payload.user.hasOwnProperty('admin')) {
```

Comment:
Just use `user.teacherId` instead of `user.hasOwnProperty`. Do the same thing for other usages of this approach

Reply:
done